v0.6.5 started failing with "var.provisioned_billing_model_version is null"

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Greenfield/Brownfield provisioning

greenfield

### Terraform Version

1.11

### Module Version

0.6.5

### AzureRM Provider Version

4.40.0

### Affected Resource(s)/Data Source(s)

Azure/avm-res-storage-storageaccount/azurerm

### Terraform Configuration Files

```hcl
NA
```

### tfvars variables values

```hcl
NA
```

### Debug Output/Panic Output

```shell
Error: Invalid function argument

  on .terraform/modules/storage_account/variables.storageaccount.tf line 229, in variable "provisioned_billing_model_version":
 229:     condition     = var.provisioned_billing_model_version == null || contains(["V2"], var.provisioned_billing_model_version)
    ├────────────────
    │ while calling contains(list, value)
    │ var.provisioned_billing_model_version is null

Invalid value for "value" parameter: argument must not be null.
Error: Terraform exited with code 1.
Error: Process completed with exit code 1.
```

### Expected Behaviour

Worked with version v0.6.4. Broken in v0.6.5

### Actual Behaviour

Terraform throws the below error,

Error: Invalid function argument

  on .terraform/modules/storage_account/variables.storageaccount.tf line 229, in variable "provisioned_billing_model_version":
 229:     condition     = var.provisioned_billing_model_version == null || contains(["V2"], var.provisioned_billing_model_version)
    ├────────────────
    │ while calling contains(list, value)
    │ var.provisioned_billing_model_version is null

Invalid value for "value" parameter: argument must not be null.
Error: Terraform exited with code 1.
Error: Process completed with exit code 1.

### Steps to Reproduce

Run terraform -chdir=${TF_WORKING_DIR} validate -no-color

### Important Factoids

_No response_

### References

_No response_

--- a/variables.storageaccount.tf
+++ b/variables.storageaccount.tf
@@ -226,7 +226,7 @@ variable "provisioned_billing_model_version" {
   description = "(Optional) Specifies the version of the provisioned billing model (e.g. when account_kind = \"FileStorage\" for Storage File). Possible value is V2. Changing this forces a new resource to be created."
 
   validation {
-    condition     = var.provisioned_billing_model_version == null || contains(["V2"], var.provisioned_billing_model_version)
+    condition     = var.provisioned_billing_model_version == null || var.provisioned_billing_model_version == "V2"
     error_message = "Invalid value for provisioned_billing_model_version. Valid options are `V2`."
   }
 }

